Assets/MixedRealityToolkit/_Core/Devices/BaseController.cs

```diff
@@ -0,0 +1,130 @@
+        public BaseController(ControllerState controllerState,
+                              Handedness controllerHandedness,
+                              IMixedRealityInputSource inputSource = null,
+                              MixedRealityInteractionMapping[] interactions = null)
+        {
+            ControllerState = controllerState;
+            ControllerHandedness = controllerHandedness;
+            InputSource = inputSource;
+            Interactions = interactions;
+        }
+
+        /// <summary>
+        /// Returns the current Input System if enabled, otherwise null.
+        /// </summary>
+        protected IMixedRealityInputSystem InputSystem
+        {
+            get
+            {
+                if (inputSystem == null && MixedRealityManager.Instance.ActiveProfile.EnableInputSystem)
+                {
+                    inputSystem = MixedRealityManager.Instance.GetManager<IMixedRealityInputSystem>();
+                }
+
+                return inputSystem;
+            }
+        }
+
+        private IMixedRealityInputSystem inputSystem;
+
+        /// <inheritdoc />
+        public ControllerState ControllerState { get; protected set; }
+
+        /// <inheritdoc />
+        public Handedness ControllerHandedness { get; }
+
+        /// <inheritdoc />
+        public IMixedRealityInputSource InputSource { get; }
+
+        /// <inheritdoc />
+        public MixedRealityInteractionMapping[] Interactions { get; protected set; }
+
+        public void SetupConfiguration(Type controllerType)
+        {
+            if (MixedRealityManager.Instance.ActiveProfile.EnableControllerProfiles)
+            {
+                // We can only enable controller profiles if mappings exist.
+                var controllerMappings = MixedRealityManager.Instance.ActiveProfile.ControllersProfile.MixedRealityControllerMappingProfiles;
+
+                for (int i = 0; i < controllerMappings?.Length; i++)
+                {
+                    if (controllerMappings[i].Controller.Type == controllerType && controllerMappings[i].Handedness == ControllerHandedness)
+                    {
+                        AssignControllerMappings(controllerMappings[i].Interactions);
+                        break;
+                    }
+                }
+
+                Debug.LogWarning($"No Controller mapping found for {controllerType}");
+            }
+        }
+
+        /// <summary>
+        /// Load the Interaction mappings for this controller from the configured Controller Mapping profile
+        /// </summary>
+        /// <param name="mappings">Configured mappings from a controller mapping profile</param>
+        protected void AssignControllerMappings(MixedRealityInteractionMapping[] mappings)
+        {
+            var interactions = new List<MixedRealityInteractionMapping>();
+            for (int i = 0; i < mappings.Length; i++)
+            {
+                switch (mappings[i].AxisType)
```

Comment:
These calls are all the same. why not just check that the AxisType is supported and throw if not... maybe something like:
if ( ((uint)mappings[i]AxisType < (uint)AxisType.None) || ((uint)mappings[i].AxisType > (uint)AxisType.SixDof))
{ throw new ArgumentOutOfRangeException(); }

Reply:
Cleaned up, these were there initially for the interactionmapping<T> method.  As we are no longer doing that, this wasn't needed anymore.